### PR TITLE
CommandsHelper class and Help module refactoring

### PR DESCRIPTION
**Pull request by mdudzinski**
This is a PR related to the refactoring discussion under #180.

### CODE REVIEW

**Comment by mdudzinski** [comment]
@dblock I took into account [your comment](https://github.com/slack-ruby/slack-ruby-bot/pull/173#issuecomment-353742851) about `CommandsHelper` naming issue.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ### 0.10.6 (Next)
 
 * Your contribution here.
+* [#182](https://github.com/slack-ruby/slack-ruby-bot/pull/182): CommandsHelper class and Help module refactoring - [@mdudzinski](https://github.com/mdudzinski).
```
Comment (dblock) [inline CHANGELOG.md]: Sounds better as `Refactored ...`

File: lib/slack-ruby-bot/commands.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 require 'slack-ruby-bot/commands/base'
 require 'slack-ruby-bot/commands/about'
-require 'slack-ruby-bot/commands/help'
+require 'slack-ruby-bot/commands/help_command'
```
Comment (dblock) [inline lib/slack-ruby-bot/commands.rb]: This one should be `help`, just like `about` or `hi`, not `_command`, no?

File: CHANGELOG.md (hunk 1 above)
Comment (mdudzinski) [reply CHANGELOG.md]: Sure, I'll change it.

File: lib/slack-ruby-bot/commands.rb (hunk 2 above)
Comment (mdudzinski) [reply lib/slack-ruby-bot/commands.rb]: Well, I didn't want it to be confusing with `SlackRubyBot::Commands::Help` from `commands/support`. And the class name is `HelpCommand` so I thought its file should match that name.

Comment (dblock) [reply lib/slack-ruby-bot/commands.rb]: I'm just thinking either all commands are `SomethingCommand` or `Something`. Thoughts?

Comment (mdudzinski) [reply lib/slack-ruby-bot/commands.rb]: Just `Something` seems to be better but then the help command's class should be renamed to `Help` and we'll have to change the name of `Help` class in support. Or add a namespace like `SlackRubyBot::Commands::Support::Help`.

Comment (dblock) [reply lib/slack-ruby-bot/commands.rb]: I like the namespace idea.

Comment (mdudzinski) [reply lib/slack-ruby-bot/commands.rb]: @dblock Okay I'll change it. If there's anything else, let me know. Thanks :)

File: CHANGELOG.md (hunk 1 above)
Comment (mdudzinski) [reply CHANGELOG.md]: Actually, it should be `Refactor ...` as commit title/summary should be imperative. Unless you want the pr's title to be different than the commit's one.

**Comment by mdudzinski** [comment]
@dblock Updated, please take a look. Thanks!

**Comment by dblock** [comment]
Perfect, merged.

**Pull request merged**